src/i965_encoder.c

```diff
@@ -1361,9 +1415,11 @@ intel_encoder_end_picture(VADriverContextP ctx,
 
     /* VME or PAK stages are separately invoked if middleware configured the corresponding
      * FEI modes through confgiruation attributes. On the other hand, ENC_PAK mode
-     * will invoke both VME and PAK similar to the non fei use case */
-    if (encoder_context->fei_enabled) {
-        if (encoder_context->fei_function_mode == VA_FEI_FUNCTION_ENC) {
+     * will invoke both VME and PAK similar to the non fei use case.
+     * PreEnc always invoke the VME */
+    if (encoder_context->fei_enabled || encoder_context->preenc_enabled) {
+        if ((encoder_context->fei_function_mode == VA_FEI_FUNCTION_ENC) ||
+            (encoder_context->preenc_enabled)) {
```

Comment:
I think if (encoder_context->preenc_enabled || ( encoder_context->fei_enabled && encoder_context->fei_function_mode == VA_FEI_FUNCTION_ENC ))  is more readable.

Reply:
No specific advantage, anyway we need to check for (fei && pak only) inside the main condition . :)

Reply:
Agree after double-checked the code, : )